src/App.js

```diff
@@ -2,20 +2,33 @@ import React from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem('savedTodoList'))
```

Comment:
If the value returned from locastorage is null (when there aren't items in the list), the app would break - because the map method can't read falsey values (null or undefined).

You can see this error in the console:

![error-elena](https://user-images.githubusercontent.com/70614187/232967408-640e945a-318b-4c5a-8983-2a4df1a0348b.png)

"Can't read properties of null" In order to prevent your app from breaking when having a falsey value, there is a trick I learned from a mentor in the React class:

value || [] is a quick shorthand for "set to empty array if value is falsey"

in other words:

React.useState(JSON.parse(localStorage.getItem('savedTodoList')) || [])